projects/aws/upgrader/upgrade/upgrader.go

```diff
@@ -0,0 +1,118 @@
+package upgrade
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/aws/eks-anywhere-build-tooling/tools/version-tracker/pkg/util/logger"
```

Comment:
What value does this library provide over just using the `logr` package directly? Also, this package has not avoided package state in its implementation. There is no reason for a logger to have package state, it's a solved problem. I recommend not using it and to make all dependencies explicit by passing them into funds and methods.